src/Compilers/Core/Portable/SymbolDisplay/AbstractSymbolDisplayVisitor_Minimal.cs

```diff
@@ -39,19 +37,7 @@ protected bool NameBoundSuccessfullyToSameSymbol(INamedTypeSymbol symbol)
             ImmutableArray<ISymbol> typeOnlySymbols = semanticModelOpt.LookupNamespacesAndTypes(positionOpt, name: symbol.Name);
             ISymbol typeOnlySymbol = SingleSymbolWithArity(typeOnlySymbols, symbol.Arity);
 
-            if (typeOnlySymbol == null)
-            {
-                return false;
-            }
-
-            var type1 = GetSymbolType(normalSymbol);
-            var type2 = GetSymbolType(typeOnlySymbol);
-
-            return
-                type1 != null &&
-                type2 != null &&
-                type1.Equals(type2) &&
-                typeOnlySymbol.Equals(symbol.OriginalDefinition);
+            return typeOnlySymbol == null ? false : typeOnlySymbol.Equals(symbol.OriginalDefinition);
```

Comment:
@AlekseyTs and I debugged through a couple of cases. He is correct that this change, that I suggested, breaks existing cases, although we don't have any tests that fail currently. We should add a test, perhaps adding the following to SemanticInfoQuickInfoSourceTests.cs.
```
        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
        public async Task PropertyWithSameNameAsOtherType()
        {
            await TestAsync(
@"namespace ConsoleApplication1
{
    class Program
    {
        static A B { get; set; }
        static B A { get; set; }

        static void Main(string[] args)
        {
            B = ConsoleApplication1.B$$.F();
        }
    }
    class A { }
    class B
    {
        public static A F() => null;
    }
}",
                MainDescription($"ConsoleApplication1.A ConsoleApplication1.B.F()"));
        }
```
We suspect that to fix the original issue, #2644, this method should be changed to always lookup symbols in a type-only context when looking up symbols for type arguments. If `AbstractSymbolDisplayVisitor` tracks when it visits type arguments, perhaps as a field on the visitor, then the first line in this method becomes `var normalSymbols = _inTypeArguments || ShouldRestrictMinimallyQualifyLookupToNamespacesAndTypes() ? ... : ...;` and the rest of the method could be reverted. (Thanks @AlekseyTs.)